### PR TITLE
Show a Recalled message for recalled items, and show recalls alongside holds

### PR DESCRIPTION
**Pull request by aleishaa**
NOTE: This depends on Bug 36075 to be pushed to Koha

https://bugs.koha-community.org/bugzilla3/show_bug.cgi?id=36075

### CODE REVIEW

**Review by demiankatz — CHANGES_REQUESTED** [review]
@aleishaa, thanks for sharing this! Would it be possible to set this up so that it doesn't fail if the recalls endpoint does not exist? It's possible that people will upgrade VuFind before upgrading their Koha instance or REST plugin, and we don't want that scenario to cause fatal errors for people. We could either add a config setting to toggle the recall retrieval, or we could put a try..catch around the recall lookup... or maybe both, in that we could create a setting to "tolerate recall failures" or something like that. What do you think?

Also interested in @EreMaijala's opinion.

**Comment by EreMaijala** [comment]
Perhaps a setting to enable recalls? Something like this in KohaRest.ini:
```
[Holds]
; Uncomment the following like to enable recalls (disabled by default). Requires a 
; Koha version that includes 
; https://bugs.koha-community.org/bugzilla3/show_bug.cgi?id=36075
;enableRecalls = true

```
This would also ensure that recalls are not checked if the library doesn't use them.

File: module/VuFind/src/VuFind/ILS/Driver/KohaRest.php [hunk 1]
```diff
@@ -173,6 +173,7 @@ class KohaRest extends \VuFind\ILS\Driver\AbstractBase implements
         'gonenoaddress' => 'patron_status_address_missing',
         'debarred' => 'patron_status_card_blocked',
         'debt' => 'renew_debt',
+        'recalled' => 'Item requested - due date changed - return on due date',
```
Comment (EreMaijala) [inline module/VuFind/src/VuFind/ILS/Driver/KohaRest.php:177]: This should be a translation code like `ILSMessages::renewal_recalled`. We are now adding ILS message translations to ILSMessages text domain, so the actual text would be in e.g. `languages/ILSMessages/en.ini`.

**Comment by demiankatz** [comment]
@aleishaa, I also noticed that this PR had a conflict with dev, so I went ahead and resolved it for you. I noticed while doing so that you seem to be using your local dev branch as the basis of this PR. I'd recommend creating a different named branch for each PR you commit, since changing a branch will also cause updates to the associated PR, and it's easy to accidentally send the wrong changes to the wrong PR if you use your dev branch as the basis. I don't think it will cause a problem here as long as we're all careful, but I'd recommend doing things differently for your next PR, if possible. :-)

**Comment by aleishaa** [comment]
thank you both for the notes! i'll close the pull request and do some better branches and work in your suggestions :)

**Comment by aleishaa** [comment]
I've created two new pull requests for the work referenced here:

- https://github.com/vufind-org/vufind/pull/3421
- https://github.com/vufind-org/vufind/pull/3422

Thanks for your suggestions!

**Pull request closed without merging**